File: suites/03_data_structures/comparisons/hashmap_operations.c
#include <stdio.h>
static long long hashmap_probe_cost(long long slots, long long rounds) {
    long long total = 0;
    for (long long r = 0; r < rounds; ++r) {
        long long key = (r * 2654435761LL) % 2147483647;
        total += ((key % slots) * 37) % 4099;
    }
    return total;
}
int main(void) { printf("%lld\n", hashmap_probe_cost(65536, 5000000)); return 0; }

File: suites/03_data_structures/comparisons/hashmap_operations.rs
fn hashmap_probe_cost(slots: i64, rounds: i64) -> i64 {
    let mut total: i64 = 0;
    for r in 0..rounds {
        let key = (r.wrapping_mul(2654435761)) % 2147483647;
        total += ((key % slots) * 37) % 4099;
    }
    total
}
fn main() { println!("{}", hashmap_probe_cost(65536, 5000000)); }
